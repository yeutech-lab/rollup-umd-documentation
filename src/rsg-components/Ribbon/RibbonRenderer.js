import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

export const styles = ({ color, space, fontSize, fontFamily }) => ({
	root: {
		position: 'fixed',
		top: 0,
		right: 0,
		width: 149,
		height: 149,
		zIndex: 999,
	},
	link: {
		fontFamily: fontFamily.base,
		position: 'relative',
		right: -37,
		top: -22,
		display: 'block',
		width: 190,
		padding: [[space[0], space[2]]],
		textAlign: 'center',
		color: color.ribbonText,
		fontSize: fontSize.base,
		background: color.ribbonBackground,
		textDecoration: 'none',
		textShadow: [[0, '-1px', 0, 'rgba(0,0,0,.15)']],
		transformOrigin: [[0, 0]],
		transform: 'rotate(45deg)',
		cursor: 'pointer',
	},
});

export function RibbonRenderer({ classes, url, text }) {
	return (
		<div className={classes.root}>
			<a href={url} className={classes.link}>
				{text}
			</a>
		</div>
	);
}

RibbonRenderer.defaultProps = {
	text: 'Fork me on GitHub',
};

RibbonRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	url: PropTypes.string.isRequired,
	text: PropTypes.string,
};

export default Styled(styles)(RibbonRenderer);
