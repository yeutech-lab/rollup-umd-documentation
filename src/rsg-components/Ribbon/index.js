export { default } from '../Ribbon/Ribbon';
