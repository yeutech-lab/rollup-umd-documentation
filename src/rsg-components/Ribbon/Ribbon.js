import React from 'react'; // eslint-disable-line no-unused-vars
import PropTypes from 'prop-types';
import RibbonRenderer from './RibbonRenderer'; // eslint-disable-line no-unused-vars

export default function Ribbon(props, { config }) {
  const { ribbon } = config;
  return ribbon ? <RibbonRenderer {...ribbon} /> : null;
}

Ribbon.contextTypes = {
  config: PropTypes.object,
};
