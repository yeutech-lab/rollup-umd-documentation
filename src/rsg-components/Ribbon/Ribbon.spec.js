import React from 'react';
import Ribbon from './Ribbon';
import { RibbonRenderer, styles } from './RibbonRenderer';

const props = {
	classes: classes(styles),
};

describe('Ribbon', () => {
	it('should render ribbon if the ribbon is present in the config', () => {
		const actual = shallow(<Ribbon />, { context: { config: { ribbon: { url: 'foo.bar' } } } });

		expect(actual).toMatchSnapshot();
	});

	it('should return null if the ribbon is not present in the config', () => {
		const actual = shallow(<Ribbon />, { context: { config: {} } });

		expect(actual.type()).toBeNull();
	});
});
describe('RibbonRenderer', () => {
	it('should render ribbon with url', () => {
		const actual = shallow(<RibbonRenderer {...props} url="http://example.com" />);

		expect(actual).toMatchSnapshot();
	});

	it('should render ribbon with a text', () => {
		const actual = shallow(
			<RibbonRenderer {...props} url="http://example.com" text="Share the repo" />
		);

		expect(actual).toMatchSnapshot();
	});
});
