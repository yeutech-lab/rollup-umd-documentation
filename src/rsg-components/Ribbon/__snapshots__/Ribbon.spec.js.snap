// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ribbon should render ribbon if the ribbon is present in the config 1`] = `
<Styled(Ribbon)
  url="foo.bar"
/>
`;

exports[`RibbonRenderer should render ribbon with a text 1`] = `
<div
  className="root"
>
  <a
    className="link"
    href="http://example.com"
  >
    Share the repo
  </a>
</div>
`;

exports[`RibbonRenderer should render ribbon with url 1`] = `
<div
  className="root"
>
  <a
    className="link"
    href="http://example.com"
  >
    Fork me on GitHub
  </a>
</div>
`;
